subsurface: 4.8.2 -> 4.9.3

flokli opened the pull request:
This bumps subsurface to the latest release, and applies a patch to make
it work with libgit2 v1.0.

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

flokli committed:
subsurface: 4.8.2 -> 4.9.3

This bumps subsurface to 4.9.3.

As the subsurface-specific libdivecomputer source is shipped as a git
submodule these days, we can avoid having to fetch a separate tarball
that needs to be kept in sync (and is only uploaded for each release).

This also applies a patch on top of 4.9.3 to properly handle libgit2
versions 1.0 and later - conditional code for older libgit versions has
been removed upstream, but that's a nontrivial backport, which is why
this patch was added.

flokli commented:
I'd like to postpone this until https://github.com/Subsurface-divelog/subsurface/pull/2737 has been merged, and maybe a release has been done.

dirkhh commented:
> I'd like to postpone this until [Subsurface-divelog/subsurface#2737](https://github.com/Subsurface-divelog/subsurface/pull/2737) has been merged, and maybe a release has been done.

I just merged the PR, but a release isn't immediately forthcoming. I want us to finish some cylinder undo code before creating a 4.10 release... and then there's updates to the user manual, endless testing, etc... so this is several weeks away at best.

flokli commented:
I fixed the off-by-one error in the original patch and added it to nixpkgs - this seemed preferrable over bumping to a WIP snapshot, or keeping subsurface broken till a release is done.

I know patches in nixpkgs are usually discouraged, but this one is really small, fixed in upstreams master in a nontrivial backportable fashion, and more understandable than something like [this](https://git.archlinux.org/svntogit/community.git/commit/trunk?h=packages/subsurface&id=e9c978cd8b0f2a5feef94ae6038ded4e0812e83e).

PTAL.
